### PR TITLE
Use ansible facts instead of facter.

### PR DESCRIPTION
**Pull request by feanil**
The ansible_processor field provides a list of all hyperthreaded cores. Using
Jinja we can just get the number of cores on the machine by running length on
this list.

### CODE REVIEW

File: playbooks/roles/edxapp/templates/cms.conf.j2 [hunk 1]
```diff
@@ -11,10 +11,8 @@ respawn limit 3 30
 env PID=/var/tmp/cms.pid
 #env NEW_RELIC_CONFIG_FILE={{app_base_dir}}/newrelic.ini
 #env NEWRELIC={{venv_dir}}/bin/newrelic-admin
-{% if facter_processorcount|int(default="NaN") is number  %}
-env WORKERS={{ worker_core_mult.cms * facter_processorcount|int }}
-{% elif ansible_processor_cores|int(default="NaN") is number %}
-env WORKERS={{ worker_core_mult.cms * ansible_processor_cores|int }}
+{% if ansible_processor > 0 %}
```
Comment (jarv) [inline playbooks/roles/edxapp/templates/cms.conf.j2]: missing `|length`?

**Comment by jbau** [comment]
yup. i think this is fine.

File: playbooks/roles/edxapp/templates/lms-xml.conf.j2 [hunk 2]
```diff
@@ -11,10 +11,8 @@ respawn limit 3 30
 env PID=/var/tmp/lms-xml.pid
 #env NEW_RELIC_CONFIG_FILE={{app_base_dir}}/newrelic.ini
 #env NEWRELIC={{venv_dir}}/bin/newrelic-admin
-{% if facter_processorcount|int(default="NaN") is number %}
-env WORKERS={{ worker_core_mult.lms_xml * facter_processorcount|int  }}
-{% elif ansible_processor_cores|int(default="NaN") is number %}
-env WORKERS={{ worker_core_mult.lms_xml * ansible_processor_cores|int  }}
+{% if ansible_processor > 0 %}
```
Comment (jarv) [inline playbooks/roles/edxapp/templates/lms-xml.conf.j2]: same comment as above

File: playbooks/roles/edxapp/templates/lms.conf.j2 [hunk 3]
```diff
@@ -9,10 +9,8 @@ respawn limit 3 30
 env PID=/var/tmp/lms.pid
 #env NEW_RELIC_CONFIG_FILE={{app_base_dir}}/newrelic.ini
 #env NEWRELIC={{venv_dir}}/bin/newrelic-admin
-{% if facter_processorcount|int(default="NaN") is number  %}
-env WORKERS={{ worker_core_mult.lms * facter_processorcount|int }}
-{% elif ansible_processor_cores|int(default="NaN") is number %}
-env WORKERS={{ worker_core_mult.lms * ansible_processor_cores|int }}
+{% if ansible_processor > 0 %}
```
Comment (jbau) [inline playbooks/roles/edxapp/templates/lms.conf.j2]: here too.

**Comment by jarv** [comment]
looks good

**Pull request merged**